### PR TITLE
haproxy test container

### PR DESCRIPTION
**Pull request by andamian**

### CODE REVIEW

File: cadc-haproxy-dev/src/haproxy/cacerts/ChainBundle2.crt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+-----BEGIN CERTIFICATE-----
```
Comment (pdowler) [inline cadc-haproxy-dev/src/haproxy/cacerts/ChainBundle2.crt]: What is ChainBundle2.crt??

File: cadc-haproxy-dev/src/haproxy/cacerts/ca.crt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```
Comment (pdowler) [inline cadc-haproxy-dev/src/haproxy/cacerts/ca.crt]: This should be renamed to cadc-internal-ca.crt in the repo

File: cadc-haproxy-dev/src/haproxy/cacerts/cadc_CA_2029.crt [hunk 3]
```diff
@@ -0,0 +1,80 @@
+Certificate:
```
Comment (pdowler) [inline cadc-haproxy-dev/src/haproxy/cacerts/cadc_CA_2029.crt]: We should haggle a bit with John about the name of this if we are going to put it into github.

File: cadc-haproxy-dev/src/haproxy/haproxy.cfg [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  option httpchk HEAD / HTTP/1.0
+
+frontend https-in
+   log global
+   bind *:443 ssl crt /conf/server-cert.pem ca-file /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem verify optional
+   mode http
+   option forwardfor
+   option http-server-close
+   http-request set-header X-Client-Certificate %[ssl_c_der,base64]
+   capture request header Origin len 128
```
Comment (pdowler) [inline cadc-haproxy-dev/src/haproxy/haproxy.cfg]: This CORS stuff (capture origin and beyond) has our internal domain name, making this not very reusable by someone who might want to contribute... 

Is there a sane way we could add this dynamically... get it from /conf/frontent.conf and stick it into the right part of the file?

**Comment by andamian** [comment]
> I wonder if the CA certs to add should just be on /conf and the startup could copy them into place before running update-ca-trust? Then we wouldn't have to care about file names or distribute our internal CAs and other people could add their own set of CA certs. Would be a simple change. like++

I thought about that but I'm worry that the cert might be mistakenly included in the repo and leaked on `github`. So much so that I'm storying mine in my home directory...

File: cadc-haproxy-dev/src/haproxy/cacerts/ca.crt (hunk 2 above)
Comment (andamian) [reply cadc-haproxy-dev/src/haproxy/cacerts/ca.crt]: I can do that but then rsync-ing with the servers won't be as simple.

**Comment by pdowler** [comment]
I don't understand. I'm saying we should not include any CA certs in git or the image and add them at runtime. I think we agree. It isn't so much a leak as CA certs (without key) are supposed to be publicly available. 

It is just that CADC does not want to or try to get other sites to take our internal CAs and add them as we don't want our internal certs usable outside CADC+CANFAR. That's why I don't want to bundle them with the image... next thing someone else will want to use it and there will be a PR with their self-signed CA cert to include. Definitely do not want that so should not include ours either.

**Pull request merged**